sqlmesh/core/loader.py

```diff
@@ -58,23 +58,17 @@ def update_model_schemas(
         if not model:
             continue
 
-        default = model_defaults.get(name)
-        kwargs = {
-            "default_schema": default.schema_ if default else None,
-            "default_catalog": default.catalog if default else None,
-        }
-        table = set_default_schema_and_catalog(name, **kwargs)
         try:
-            model.update_schema(schema, **kwargs)
+            model.update_schema(schema, default_catalog=default_catalog)
             optimized_query_cache.with_optimized_query(model)
 
             columns_to_types = model.columns_to_types
             if columns_to_types is not None:
-                schema.add_table(table, columns_to_types, dialect=model.dialect)
+                schema.add_table(model.fqn, columns_to_types, dialect=model.dialect)
```

Comment:
this needs to be double checked that we aren't doing some kind of double normalization here, add_table does normalization

Reply:
I think this is as simple as passing in `normalize=False` when calling `add_table` which looks like it parses out the identifier but then doesn't normalize it after which would prevent the double normalization. Let me know if I am missing anything.